grakn-kb/src/main/java/ai/grakn/kb/internal/concept/RelationshipImpl.java

```diff
@@ -121,11 +121,8 @@ public Relationship attributeRelationship(Attribute attribute) {
 
     @Override
     public Stream<Attribute<?>> keys(AttributeType[] attributeTypes) {
-        if(reified().isPresent()){
-            return reified().get().attributes(attributeTypes);
-        } else {
-            return Stream.empty();
-        }
+        Optional<RelationshipReified> reified = reified();
+        return reified.map(relationshipReified -> relationshipReified.attributes(attributeTypes)).orElseGet(Stream::empty);
```

Comment:
You should totally one-liner this: `return reified().map(...)` - only two extra characters!